components/DateTimePicker.js

```diff
@@ -13,19 +13,40 @@ import { View } from './View';
 
 const isIos = Platform.OS === 'ios';
 
+const MODES = { DATE: 'date', DATETIME: 'datetime', TIME: 'time' };
+
+const DISPLAY_MODES = {
+  [MODES.DATE]: 'inline',
+  [MODES.DATETIME]: 'inline',
+  [MODES.TIME]: 'spinner',
+};
+
+function getDisplayMode(mode) {
+  if (!isIos || !Object.keys(DISPLAY_MODES).includes(mode)) {
+    return 'default';
+  }
+
+  return DISPLAY_MODES[mode];
+}
+
 class DateTimePicker extends PureComponent {
   constructor(props) {
     super(props);
 
     autoBindReact(this);
 
+    const { mode, value } = props;
+
     this.state = {
+      display: getDisplayMode(mode),
       showPicker: false,
-      value: props.value,
+      showTimePicker: false,
+      value,
     };
   }
 
   componentDidUpdate() {
+    // todo: check why set state here
```

Comment:
there was some buggy behaviour with this in the past, which is why we needed this piece of dirt. You can check it out and remove if possible ( it was android only issue )

Reply:
In my example app everything worked fine on Android, but since I'm not familiar with all use cases of this component, I would maybe keep it here for now

Reply:
Let's do 1 more check before merging this - switch simulator/device to dark mode & test this.
I remember there was some weird behavior that was happening only if device had dark mode/theme (other than picker not being visible)